packages/sdk/src/oracle/utils.ts

```diff
@@ -17,86 +17,84 @@ import { ethers } from "ethers";
 export const getAddress = ethers.utils.getAddress;
 export const hexValue = ethers.utils.hexValue;
 
+export function ignoreError<X extends () => any>(call: X): ReturnType<X> | undefined {
+  try {
+    return call();
+  } catch (err) {
+    return undefined;
+  }
+}
+
 export function initFlags(): Flags {
   return {
     [Flag.MissingRequest]: false,
     [Flag.MissingUser]: false,
     [Flag.WrongChain]: false,
-    [Flag.InProposeState]: false,
-    [Flag.InDisputeState]: false,
+    [Flag.CanPropose]: false,
+    [Flag.CanDispute]: false,
+    [Flag.CanSettle]: false,
+    [Flag.InDvmVote]: false,
+    [Flag.RequestSettled]: false,
     [Flag.InsufficientBalance]: false,
     [Flag.InsufficientApproval]: false,
-    [Flag.ProposalInProgress]: false,
-    [Flag.ApprovalInProgress]: false,
-    [Flag.DisputeInProgress]: false,
+    [Flag.ProposalTxInProgress]: false,
+    [Flag.ApprovalTxInProgress]: false,
+    [Flag.DisputeTxInProgress]: false,
     [Flag.ChainChangeInProgress]: false,
   };
 }
 
+export const nowS = (now = Date.now()): number => Math.floor(now / 1000);
+
+// reduce global state into important UI boolean states. this should never throw errors.
 export function getFlags(state: State): Record<Flag, boolean> {
   const read = new Read(state);
   const flags = initFlags();
 
-  try {
-    read.userAddress();
-    flags[Flag.MissingUser] = false;
-  } catch (err) {
-    flags[Flag.MissingUser] = true;
-  }
+  const signer = ignoreError(read.signer);
+  flags[Flag.MissingUser] = signer ? false : true;
 
-  try {
-    read.inputRequest();
-    flags[Flag.MissingRequest] = false;
-  } catch (err) {
-    flags[Flag.MissingRequest] = true;
-  }
+  const inputRequest = ignoreError(read.inputRequest);
+  flags[Flag.MissingRequest] = inputRequest ? false : true;
 
-  try {
-    flags[Flag.WrongChain] = read.userChainId() !== read.requestChainId();
-  } catch (err) {
-    flags[Flag.WrongChain] = false;
-  }
+  const userChainId = ignoreError(read.userChainId);
+  const requestChainId = ignoreError(read.requestChainId);
+  flags[Flag.WrongChain] = userChainId && requestChainId ? userChainId !== requestChainId : false;
 
-  try {
-    flags[Flag.InProposeState] = read.request()?.state === RequestState.Requested;
-  } catch (err) {
-    flags[Flag.InProposeState] = false;
-  }
+  const request = ignoreError(read.request);
 
-  try {
-    flags[Flag.InDisputeState] = read.request()?.state === RequestState.Proposed;
-  } catch (err) {
-    flags[Flag.InDisputeState] = false;
-  }
+  // these are a bit redundant with request state, but just an alternate way to see current request state
+  flags[Flag.CanPropose] = request?.state === RequestState.Requested;
+  flags[Flag.CanDispute] = request?.state === RequestState.Proposed;
```

Comment:
previous bug, this dispute flag could get out of date becuase it should also take into consideration expiration time. rather than add the logic here, decided rather to poll the request, updating from chain on an interval, that way this flag logic can remain simple and based on the currently known request state.  this is to prevent an edge case where we might disable dispute button even tho request may be disputable